p2panda-net/src/sync/manager.rs

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use anyhow::Result;
+use iroh_gossip::proto::TopicId;
+use iroh_net::{Endpoint, NodeId};
+use p2panda_sync::traits::SyncProtocol;
+use p2panda_sync::SyncError;
+use tokio::sync::mpsc::Sender;
+use tracing::{debug, warn};
+
+use crate::engine::ToEngineActor;
+use crate::sync::{self, SYNC_CONNECTION_ALPN};
+
+#[derive(Debug)]
+pub struct SyncManager {
+    known_peer_topics: HashMap<NodeId, HashSet<TopicId>>,
```

Comment:
When we get into our upcoming cleanup phase, I'd like to take a close look at our address book story and see if there's potential for consolidation. We currently have a peer map in the engine actor and a topic map here...they record different information but it might be nice to have a single location for look-up. Just flagging that here for now.